README.txt
This game requires the news.txt and stock_game.py files to be within the same file directory to run.
Do ensure that this is the case before executing the stock_game.py file.



CTD 1D Submissions group C09-E/news.txt
Reports of a new virus COVID-19 is starting to emerge from China (31 Dec 2018)

Things that start small can reshape the world. Few people noticed when news emerged last December that China had begun monitoring the
outbreak of a new pneumonia-like virus, or even on January 11 after China reported its first death from the disease. Nearly a year
later, COVID-19 had changed life as we knew it. The World Health Organization estimated in October that as much as 10 percent of the
world’s population had already contracted COVID-19; by year’s end some 1.7 million people had died from it. As countries implemented
lockdowns to stop the disease’s spread, the global economy sputtered—it may have contracted by more than 4 percent—and poverty rates spiked.
Some countries like New Zealand and Vietnam handled the pandemic well, limiting infections while sustaining economic growth. But many
countries, most notably the United States, bungled their response and saw soaring caseloads and death rates. Arguments will persist for
years over why the crisis was mishandled; expect poor leadership, political partisanship, and distrust of government to figure prominently
in explanations. The year closed on a positive note as vaccines were approved in record time. The challenges now are distributing them
widely and equitably while preparing to do a better job stopping the next pandemic. The novel coronavirus will not be the last of its kind.

+

Tensions flair between US and Iran (May 2019)

Bad blood between Tehran and Washington still persists after four decades. On January 3, a U.S. drone strike killed Qasem Soleimani,
the leader of the Quds Force of the Islamic Revolutionary Guard Corps, shortly after he arrived in Baghdad. Iran retaliated over the
next two months with rocket attacks on U.S. bases in Iraq, injuring dozens of U.S. troops and killing many Iraqis. Tensions flared again
in April when several Iranian speedboats harassed U.S. warships in the Persian Gulf, prompting Trump to tweet that he had issued
orders “to shoot down” Iranian gunboats if the harassment continued. Iran responded by threatening to destroy U.S. warships. Amidst this
spat, Iran launched its first military satellite, intensifying U.S. concerns over Iran’s improving long-range missile capabilities.
While the threat of direct military confrontation eased over the summer, the United States imposed new sanctions on Iran as part of
its “maximum pressure” campaign, ignoring calls from European allies countries to ease its economic pressure in light of the COVID-19
pandemic. Tensions rose again in late November when Iran’s top nuclear scientist was assassinated, likely by Israel. Iran responded by
moving even further away from the terms of the 2015 nuclear deal. The potential that Iran could soon leave the deal entirely set the stage
for high stakes diplomacy early in 2021.

+

US China trade wars continues (23 Dec 2019)

Back in March 2018, President Donald Trump tweeted that “trade wars are good and easy to win.” That hasn’t been the case so far with his
trade war with China. In February, Trump delayed imposing a new round of tariffs on Chinese goods in a bid to give negotiators time to
strike a deal. In May, he concluded that the talks hadn’t progressed and imposed a new round of tariffs. In June, Trump and Chinese President
Xi Jinping agreed at the G-20 summit to push ahead with trade talks. In August, however, Trump announced he would place tariffs on $300 billion
worth of Chinese goods effective September 1. Twelve days later, he partially reversed course, saying he would delay half those tariffs until
December 15. China wasn’t mollified; on August 23 it announced tariffs on $75 billion worth of U.S. goods. Negotiators continued to meet, and
on October 11 they reached tentative agreement on a “Phase 1” deal. It took another two months, though, before the details were worked out.
Both sides claimed victory. However, the agreement failed to settle the major differences between the two economic superpowers, suggesting
that 2020 could be turbulent on the trade front.

+

Fed cuts rates to 0 (16 Mar 2020)

The Federal Reserve, saying “the coronavirus outbreak has harmed communities and disrupted economic activity in many countries, including the
United States,” cut interest rates to essentially zero on Sunday and launched a massive $700 billion quantitative easing program to shelter
the economy from the effects of the virus. The new fed funds rate, used as a benchmark both for short-term lending for financial institutions
and as a peg to many consumer rates, will now be targeted at 0% to 0.25% down from a previous target range of 1% to 1.25%.

+

Moderna vaccine gets approval (dec 2020)

On December 18, 2020, the U.S. Food and Drug Administration issued an emergency use authorization (EUA) for Moderna COVID-19 Vaccine for the
prevention of coronavirus disease 2019 (COVID-19) caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) in individuals 18
years of age and older. Moderna COVID-19 Vaccine is available under EUA as a two-dose primary series in individuals 18 years of age and older,
as a third primary series dose for individuals 18 years of age and older who have been determined to have certain kinds of immunocompromise, and
as a single booster dose in people 18 years of age and older. The Moderna COVID-19 Vaccine is also authorized for use as a heterologous
(or “mix and match”) booster dose following completion of primary vaccination with a different available COVID-19 vaccine. For example,
Pfizer-BioNTech COVID-19 Vaccine and Janssen COVID-19 vaccine recipients 18 years of age and older may receive a single booster dose of
the Moderna COVID-19 Vaccine. 

+

This is the final turn of the game.
Do sell all of the stocks you are currently holding and press the "End Game" Button!

+

This is a turn-based game single-player game which aims to test your ability to invest in stocks based on current affairs.

Python Testing/test2.py
from tkinter import *
from tkinter import ttk

#Defining Variables

capital = 10000
stk1 = {"name": "Stock 1", "quantity": 0, "price": 100}
stk2 = {"name": "Stock 2", "quantity": 0, "price": 200}
stk3 = {"name": "Stock 3", "quantity": 0, "price": 300}
stk4 = {"name": "Stock 4", "quantity": 0, "price": 400}

#Defining Functions

def buy(stk):
   global capital
   if capital < stk["price"]:  
      print("You don't have enough capital")
   else: 
      stk["quantity"]+=1
      capital -= stk["price"]

def sell(stk):
   global capital
   if stk["quantity"]<=0:
      print("You don't own this stock")
   else:
      stk["quantity"]-=1
      capital += stk["price"]

#Creating Table

root = Tk()

content = ttk.Frame(root, padding=(3,3,12,12))
frame = ttk.Frame(content, borderwidth=5, relief="ridge", width=200, height=100)
namelbl = ttk.Label(content, text="Name")
name = ttk.Entry(content)

ok = ttk.Button(content, text="Okay")
cancel = ttk.Button(content, text="Cancel")

content.grid(column=0, row=0, sticky=(N, S, E, W))
frame.grid(column=0, row=0, columnspan=3, rowspan=2, sticky=(N, S, E, W))
namelbl.grid(column=3, row=0, columnspan=2, sticky=(N, W), padx=5)
name.grid(column=3, row=1, columnspan=2, sticky=(N,E,W), pady=5, padx=5)
ok.grid(column=3, row=3)
cancel.grid(column=4, row=3)

root.columnconfigure(0, weight=1)
root.rowconfigure(0, weight=1)
content.columnconfigure(0, weight=3)
content.columnconfigure(1, weight=3)
content.columnconfigure(2, weight=3)
content.columnconfigure(3, weight=1)
content.columnconfigure(4, weight=1)
content.rowconfigure(1, weight=1)

root.mainloop()

Python Testing/main GUI.py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:





# In[1]:


from tkinter import *
from  tkinter import ttk


capital = 10000
stk1 = {"name": "Stock 1", "quantity": 0, "price": 100}
stk2 = {"name": "Stock 2", "quantity": 0, "price": 200}
stk3 = {"name": "Stock 3", "quantity": 0, "price": 300}
stk4 = {"name": "Stock 4", "quantity": 0, "price": 400}

def buy(stk):
   global capital
   if capital < stk["price"]:  
      print("You don't have enough capital")
   else: 
      stk["quantity"]+=1
      capital -= stk["price"]

def sell(stk):
   global capital
   if stk["quantity"]<=0:
      print("You don't own this stock")
   else:
      stk["quantity"]-=1
      capital += stk["price"]

ws  = Tk()
ws.title('Stonks Game')
ws.geometry('500x500')
ws['bg'] = '#AC99F2'

game_frame = Frame(ws)
game_frame.pack()

my_game = ttk.Treeview(game_frame)

my_game['columns'] = ('player_id', 'player_name', 'player_Rank', 'player_states', 'player_city')

my_game.column("#0", width=0,  stretch=NO)
my_game.column("player_id",anchor=CENTER, width=130)
my_game.column("player_name",anchor=CENTER,width=130)
my_game.column("player_Rank",anchor=CENTER,width=130)
my_game.column("player_states",anchor=CENTER,width=130)


my_game.heading("#0",text="",anchor=CENTER)
my_game.heading("player_id",text="Stock Name",anchor=CENTER)
my_game.heading("player_name",text="Price",anchor=CENTER)
my_game.heading("player_Rank",text="Quantity",anchor=CENTER)
my_game.heading("player_states",text="Choice",anchor=CENTER)


my_game.insert(parent='',index='end',iid=0,text='',
values=(stk1["name"],stk1["price"],stk1["quantity"],))
my_game.insert(parent='',index='end',iid=1,text='',
values=(stk2["name"],stk2["price"],stk2["quantity"],))
my_game.insert(parent='',index='end',iid=2,text='',
values=(stk3["name"],stk3["price"],stk3["quantity"]))

my_game.pack()



#Buy Buttons       
# Initialize tkinter window with dimensions 300 x 250             
ws.geometry('300x250')     
  
# Creating a Button
btn = Button(ws, text = 'Buy', command = buy(stk1))
  
# Set the position of button to coordinate (200, 20)
btn.place(x=800, y=20)
btn.config(height=1,width=2)

#Button2
ws.geometry('300x250')     
  
# Creating a Button
btn2 = Button(ws, text = 'Buy', command = buy(stk2))
  
# Set the position of button to coordinate (200, 20)
btn2.place(x=800, y=40)
btn2.config(height=1,width=2)

#Button3
ws.geometry('300x250')     
  
# Creating a Button
btn2 = Button(ws, text = 'Buy', command = buy(stk3))
  
# Set the position of button to coordinate (200, 20)
btn2.place(x=800, y=60)
btn2.config(height=1,width=2)

#Sell button
ws.geometry('300x250')     
  
# Creating a Button
btn2 = Button(ws, text = 'Sell', command = sell(stk1))
  
# Set the position of button to coordinate (200, 20)
btn2.place(x=900, y=60)
btn2.config(height=1,width=2)

#Sell button2
ws.geometry('300x250')     
  
# Creating a Button
btn2 = Button(ws, text = 'Sell', command = sell(stk2))
  
# Set the position of button to coordinate (200, 20)
btn2.place(x=900, y=40)
btn2.config(height=1,width=2)

#Sell button3
ws.geometry('300x250')     
  
# Creating a Button
btn2 = Button(ws, text = 'Sell', command = sell(stk3))
  
# Set the position of button to coordinate (200, 20)
btn2.place(x=900, y=20)
btn2.config(height=1,width=2)


ws.mainloop()


# In[ ]:





# In[ ]:






Game Files/stock game old.py
#-------Importing tkinter for GUI-------
from tkinter import *
from tkinter import ttk
import os
import sys

#-------Reading news from text file-------
with open(os.path.join(sys.path[0], 'news.txt')) as f:
   nstr=f.read()
   nlist=nstr.split("+")

#-------Defining Variables-------
capital = 1000
stk1 = {"name": "Tesla", "quantity": 0, "price": 100}
stk2 = {"name": "Alibaba", "quantity": 0, "price": 100}
stk3 = {"name": "Singapore Airlines", "quantity": 0, "price": 100}
stk4 = {"name": "ExxonMobil", "quantity": 0, "price": 100}
error_msg = ""
storycounter = -1

#-------Defining Functions to be assigned to buttons-------

'''The buy function takes the stock and the stock quantity labels as input parameters.
If the user has less capital than the price of the stock, then the global error_msg variable is
changed accordingly and the error label is updated.
If the user has sufficient capital, the quantity of the stock increases by 1 and the global variable
capital is reduced by the price of the stock. The stock quantity and capital labels are updated and
the error label is cleared.'''

def buy(stk,stkq):
   global capital
   global error_msg
   if capital < stk["price"]: 
      error_msg = "Error: you don't have enough capital"
      errorlbl.config(text=error_msg)
   else: 
      stk["quantity"]+=1
      stkq.config(text=stk["quantity"])
      capital -= stk["price"]
      capitallbl.config(text="Capital:"+str(capital))
      error_msg = ""
      errorlbl.config(text=error_msg)

'''The sell function takes the stock and the stock quantity labels as input parameters.
If the user does not have any of the specified stock (i.e. its quantity is 0), then the global error_msg
variable is changed accordingly and the error label is updated.
If the user owns the stock, the quantity of the stock is reduced by 1 and the global variable capital is
increased by the price of the stock. The stock quantity and capital labels are updated and the error
label is cleared.'''

def sell(stk,stkq):
   global capital
   global error_msg
   if stk["quantity"]<=0:
      error_msg = "Error: you don't own this stock"
      errorlbl.config(text=error_msg)
   else:
      stk["quantity"]-=1
      stkq.config(text=stk["quantity"])
      capital += stk["price"]
      capitallbl.config(text="Capital:"+str(capital))
      error_msg = ""
      errorlbl.config(text=error_msg)

'''The next_turn function takes no input parameters. However, it uses the global storycounter variable
to determine what set of stock prices and news should be shown. It then increments the storycounter by 1.
Once the storycounter reaches 4, it updates the text displayed on the button from “next turn” to “end game”.
Once the storycounter reaches 5, it clears the frame and displays some text to indicate to the user that they
have finished the game.'''

def next_turn():
   global storycounter
   global stk1,stk2,stk3,stk4
   storycounter+=1
   if storycounter <4:

      stk1["price"]=data[storycounter][1]
      stk2["price"]=data[storycounter][2]
      stk3["price"]=data[storycounter][3]
      stk4["price"]=data[storycounter][4]

      news.config(text=data[storycounter][0])
      stk1p.config(text="$"+str(stk1["price"]))
      stk2p.config(text="$"+str(stk2["price"]))
      stk3p.config(text="$"+str(stk3["price"]))
      stk4p.config(text="$"+str(stk4["price"]))

      error_msg = ""
      errorlbl.config(text=error_msg)

   elif storycounter==4:

      stk1["price"]=data[storycounter][1]
      stk2["price"]=data[storycounter][2]
      stk3["price"]=data[storycounter][3]
      stk4["price"]=data[storycounter][4]

      news.config(text=data[storycounter][0])
      stk1p.config(text="$"+str(stk1["price"]))
      stk2p.config(text="$"+str(stk2["price"]))
      stk3p.config(text="$"+str(stk3["price"]))
      stk4p.config(text="$"+str(stk4["price"]))
      next.config(text="End Game")

      error_msg = ""
      errorlbl.config(text=error_msg)

   else:
      clearFrame(content)
      endlbl = ttk.Label(content, text="Congratulations, you finished the game with $"+str(capital))
      perflbl = ttk.Label(content, text=check_capital())
      tylbl = ttk.Label(content, text= "Thank you for playing our game!")
      closebtn = ttk.Button(content, text="Close", command=root.destroy)
      endlbl.grid(column=0,row=0,sticky=(N, S, E, W))
      perflbl.grid(column=0,row=1,sticky=(N, S, E, W))
      tylbl.grid(column=0,row=2,sticky=(N, S, E, W))
      closebtn.grid(column=0,row=3,sticky=(N, S, E, W))

'''The check_capital function takes no input parameters. It checks the global capital variable and returns a
message depending on how much capital is left at the end of the game. This message lets the user know how they
have performed. This function is used within the next_turn function
'''

def check_capital():
   if capital>3000:
      return "You did an amazing job, more than tripling your starting capital"
   elif 2000<capital<=3000:
      return "You did a great job, more than doubling your starting capital"
   elif 1000<capital<=2000:
      return "You did a good job, you managed to increase your capital"
   elif capital == 1000:
      return "You managed to maintain your capital"
   else:
      return "Unfortunately, you lost money"

'''The clearFrame function uses a for loop to clear the widgets within a frame. This function is used within the
next_turn function.'''

def clearFrame(frame):
   # destroy all widgets from frame
   for widget in frame.winfo_children():
      widget.destroy()

#-------Stored Text and Data-------
data = [[nlist[1],100,85,95,80],
        [nlist[2],55,90,90,85],
        [nlist[3],150,100,85,80],
        [nlist[4],180,105,50,50],
        [nlist[5],705,110,50,55]]

#-------An instance of a Tkinter Window-------
root = Tk()
root.title("Stock Game")

#-------Creating Frames-------
content = ttk.Frame(root, padding=(5,5,12,12))
frame = ttk.Frame(content, borderwidth=5, relief="ridge", width=200, height=100)

#-------Creating Labels-------
news = ttk.Label(content, text=nlist[0])
namelbl = ttk.Label(content, text="Name")
pricelbl = ttk.Label(content, text="Price")
quantlbl = ttk.Label(content, text="Quantity")
choicelbl = ttk.Label(content, text="Choice")
stk1n = ttk.Label(content, text=stk1["name"])
stk2n = ttk.Label(content, text=stk2["name"])
stk3n = ttk.Label(content, text=stk3["name"])
stk4n = ttk.Label(content, text=stk4["name"])
stk1p = ttk.Label(content, text="$"+str(stk1["price"]))
stk2p = ttk.Label(content, text="$"+str(stk2["price"]))
stk3p = ttk.Label(content, text="$"+str(stk3["price"]))
stk4p = ttk.Label(content, text="$"+str(stk4["price"]))
stk1q = ttk.Label(content, text=stk1["quantity"])
stk2q = ttk.Label(content, text=stk2["quantity"])
stk3q = ttk.Label(content, text=stk3["quantity"])
stk4q = ttk.Label(content, text=stk4["quantity"])
errorlbl = ttk.Label(content, text=error_msg)
capitallbl= ttk.Label(content, text="Capital:"+str(capital))

#-------Creating Buttons-------
buy1 = ttk.Button(content, text="Buy", command=lambda: buy(stk1,stk1q))
buy2 = ttk.Button(content, text="Buy", command=lambda: buy(stk2,stk2q))
buy3 = ttk.Button(content, text="Buy", command=lambda: buy(stk3,stk3q))
buy4 = ttk.Button(content, text="Buy", command=lambda: buy(stk4,stk4q))

sell1 = ttk.Button(content, text="Sell", command=lambda: sell(stk1,stk1q))
sell2 = ttk.Button(content, text="Sell", command=lambda: sell(stk2,stk2q))
sell3 = ttk.Button(content, text="Sell", command=lambda: sell(stk3,stk3q))
sell4 = ttk.Button(content, text="Sell", command=lambda: sell(stk4,stk4q))

next= ttk.Button(content, text="Next Turn", command=lambda: next_turn())

#-------Establishing the Grid for the buttons and labels-------
content.grid(column=0, row=0, sticky=(N, S, E, W))

#-------Placing the News and a Frame around it-------
frame.grid(column=0, row=0, columnspan=5, rowspan=1, sticky=(N, S, E, W))
news.grid(column=0, row=0, columnspan=5, rowspan =1, sticky=(N, S, E, W), pady=20, padx=20)

#-------Creating the table for the stocks-------
namelbl.grid(column=0,row=1,sticky=(N, S, E, W))
stk1n.grid(column=0,row=2,sticky=(N, S, E, W))
stk2n.grid(column=0,row=3,sticky=(N, S, E, W))
stk3n.grid(column=0,row=4,sticky=(N, S, E, W))
stk4n.grid(column=0,row=5,sticky=(N, S, E, W))

pricelbl.grid(column=1,row=1,sticky=(N, S, E, W))
stk1p.grid(column=1,row=2,sticky=(N, S, E, W))
stk2p.grid(column=1,row=3,sticky=(N, S, E, W))
stk3p.grid(column=1,row=4,sticky=(N, S, E, W))
stk4p.grid(column=1,row=5,sticky=(N, S, E, W))

quantlbl.grid(column=2,row=1,sticky=(N, S, E, W))
stk1q.grid(column=2,row=2,sticky=(N, S, E, W))
stk2q.grid(column=2,row=3,sticky=(N, S, E, W))
stk3q.grid(column=2,row=4,sticky=(N, S, E, W))
stk4q.grid(column=2,row=5,sticky=(N, S, E, W))

choicelbl.grid(column=3,row=1,columnspan=2,sticky=(N, S, E, W))

buy1.grid(column=3,row=2,sticky=(N, S, E, W))
buy2.grid(column=3,row=3,sticky=(N, S, E, W))
buy3.grid(column=3,row=4,sticky=(N, S, E, W))
buy4.grid(column=3,row=5,sticky=(N, S, E, W))

sell1.grid(column=4,row=2,sticky=(N, S, E, W))
sell2.grid(column=4,row=3,sticky=(N, S, E, W))
sell3.grid(column=4,row=4,sticky=(N, S, E, W))
sell4.grid(column=4,row=5,sticky=(N, S, E, W))

#-------Placing the labels/button at the bottom-------
errorlbl.grid(column=0,row=6,columnspan=3,sticky=(N, S, E, W))
capitallbl.grid(column=3,row=6,columnspan=1,sticky=(N, S, E, W))
next.grid(column=4,row=6,columnspan=1,sticky=(N, S, E, W))

#-------Configuring the columns and rows to scale-------
root.columnconfigure(0, weight=1)
root.rowconfigure(0, weight=1)

content.rowconfigure(0, weight=6)
for i in range(5):
   content.columnconfigure(i, weight=1)
for i in range(1,6):
   content.rowconfigure(i, weight=1)

#-------Running the window-------
root.mainloop()

CTD 1D Submissions group C09-E/stock game.py
#-------Importing tkinter for GUI-------
from tkinter import *
from tkinter import ttk

#-------Reading news from text file-------
with open('news.txt') as f:
   nstr=f.read()
   nlist=nstr.split("+")

#-------Defining Variables-------
capital = 1000
stk1 = {"name": "Tesla", "quantity": 0, "price": 100}
stk2 = {"name": "Alibaba", "quantity": 0, "price": 100}
stk3 = {"name": "Singapore Airlines", "quantity": 0, "price": 100}
stk4 = {"name": "ExxonMobil", "quantity": 0, "price": 100}
error_msg = ""
storycounter = -1

#-------Defining Functions to be assigned to buttons-------

'''The buy function takes the stock and the stock quantity labels as input parameters.
If the user has less capital than the price of the stock, then the global error_msg variable is
changed accordingly and the error label is updated.
If the user has sufficient capital, the quantity of the stock increases by 1 and the global variable
capital is reduced by the price of the stock. The stock quantity and capital labels are updated and
the error label is cleared.'''

def buy(stk,stkq):
   global capital
   global error_msg
   if capital < stk["price"]: 
      error_msg = "Error: you don't have enough capital"
      errorlbl.config(text=error_msg)
   else: 
      stk["quantity"]+=1
      stkq.config(text=stk["quantity"])
      capital -= stk["price"]
      capitallbl.config(text="Capital: $"+str(capital))
      error_msg = ""
      errorlbl.config(text=error_msg)

'''The sell function takes the stock and the stock quantity labels as input parameters.
If the user does not have any of the specified stock (i.e. its quantity is 0), then the global error_msg
variable is changed accordingly and the error label is updated.
If the user owns the stock, the quantity of the stock is reduced by 1 and the global variable capital is
increased by the price of the stock. The stock quantity and capital labels are updated and the error
label is cleared.'''

def sell(stk,stkq):
   global capital
   global error_msg
   if stk["quantity"]<=0:
      error_msg = "Error: you don't own this stock"
      errorlbl.config(text=error_msg)
   else:
      stk["quantity"]-=1
      stkq.config(text=stk["quantity"])
      capital += stk["price"]
      capitallbl.config(text="Capital: $"+str(capital))
      error_msg = ""
      errorlbl.config(text=error_msg)

'''The next_turn function takes no input parameters. However, it uses the global storycounter variable
to determine what set of stock prices and news should be shown. It then increments the storycounter by 1.
Once the storycounter reaches 4, it updates the text displayed on the button from “next turn” to “end game”.
Once the storycounter reaches 5, it clears the frame and displays some text to indicate to the user that they
have finished the game.'''

def next_turn():
   global storycounter
   global stk1,stk2,stk3,stk4
   storycounter+=1
   if storycounter <4:

      stk1["price"]=data[storycounter][1]
      stk2["price"]=data[storycounter][2]
      stk3["price"]=data[storycounter][3]
      stk4["price"]=data[storycounter][4]

      news.config(text=data[storycounter][0])
      stk1p.config(text="$"+str(stk1["price"]))
      stk2p.config(text="$"+str(stk2["price"]))
      stk3p.config(text="$"+str(stk3["price"]))
      stk4p.config(text="$"+str(stk4["price"]))

      error_msg = ""
      errorlbl.config(text=error_msg)

   elif storycounter==4:

      stk1["price"]=data[storycounter][1]
      stk2["price"]=data[storycounter][2]
      stk3["price"]=data[storycounter][3]
      stk4["price"]=data[storycounter][4]

      news.config(text=data[storycounter][0])
      stk1p.config(text="$"+str(stk1["price"]))
      stk2p.config(text="$"+str(stk2["price"]))
      stk3p.config(text="$"+str(stk3["price"]))
      stk4p.config(text="$"+str(stk4["price"]))
      next.config(text="End Game")

      error_msg = ""
      errorlbl.config(text=error_msg)

   else:
      clearFrame(content)
      endlbl = ttk.Label(content, text="Congratulations, you finished the game with $"+str(capital))
      perflbl = ttk.Label(content, text=check_capital())
      tylbl = ttk.Label(content, text= "Thank you for playing our game!")
      closebtn = ttk.Button(content, text="Close", command=root.destroy)
      endlbl.grid(column=0,row=0,sticky=(N, S, E, W))
      perflbl.grid(column=0,row=1,sticky=(N, S, E, W))
      tylbl.grid(column=0,row=2,sticky=(N, S, E, W))
      closebtn.grid(column=0,row=3,sticky=(N, S, E, W))

'''The check_capital function takes no input parameters. It checks the global capital variable and returns a
message depending on how much capital is left at the end of the game. This message lets the user know how they
have performed. This function is used within the next_turn function
'''

def check_capital():
   if capital>3000:
      return "You did an amazing job, more than tripling your starting capital."
   elif 2000<capital<=3000:
      return "You did a great job, more than doubling your starting capital."
   elif 1000<capital<=2000:
      return "You did a good job, you managed to increase your capital."
   elif capital == 1000:
      return "You managed to maintain your capital."
   else:
      return "Unfortunately, you lost money."

'''The clearFrame function uses a for loop to clear the widgets within a frame. This function is used within the
next_turn function.'''

def clearFrame(frame):
   # destroy all widgets from frame
   for widget in frame.winfo_children():
      widget.destroy()

#-------Stored Text and Data-------
data = [[nlist[1],100,85,95,80],
        [nlist[2],55,90,90,85],
        [nlist[3],150,100,85,80],
        [nlist[4],180,105,50,50],
        [nlist[5],705,110,50,55]]

#-------Creating the initial game window-------
master = Tk()
master.title("Stock Game")
titlelbl =Label(master, text="Welcome to the stock game!")
desclbl =Label(master, text=nlist[6])

closebtn=ttk.Button(master, text="Play", command=master.destroy)
titlelbl.grid(column=0,row=0,sticky=(N, S, E, W))
desclbl.grid(column=0,row=1,sticky=(N, S, E, W))
closebtn.grid(column=0,row=2,sticky=(N, S, E, W))
master.rowconfigure(0, weight=1)
master.rowconfigure(1, weight=5)
master.rowconfigure(2, weight=1)
master.columnconfigure(0,weight=1)

master.mainloop()

#-------An instance of a Tkinter Window-------
root = Tk()
root.title("Stock Game")

#-------Creating Frames-------
content = ttk.Frame(root, padding=(5,5,12,12))
frame = ttk.Frame(content, borderwidth=5, relief="ridge", width=200, height=100)

#-------Creating Labels-------
news = ttk.Label(content, text=nlist[0])
namelbl = ttk.Label(content, text="Name")
pricelbl = ttk.Label(content, text="Price")
quantlbl = ttk.Label(content, text="Quantity")
choicelbl = ttk.Label(content, text="Choice")
stk1n = ttk.Label(content, text=stk1["name"])
stk2n = ttk.Label(content, text=stk2["name"])
stk3n = ttk.Label(content, text=stk3["name"])
stk4n = ttk.Label(content, text=stk4["name"])
stk1p = ttk.Label(content, text="$"+str(stk1["price"]))
stk2p = ttk.Label(content, text="$"+str(stk2["price"]))
stk3p = ttk.Label(content, text="$"+str(stk3["price"]))
stk4p = ttk.Label(content, text="$"+str(stk4["price"]))
stk1q = ttk.Label(content, text=stk1["quantity"])
stk2q = ttk.Label(content, text=stk2["quantity"])
stk3q = ttk.Label(content, text=stk3["quantity"])
stk4q = ttk.Label(content, text=stk4["quantity"])
errorlbl = ttk.Label(content, text=error_msg)
capitallbl= ttk.Label(content, text="Capital: $"+str(capital))

#-------Creating Buttons-------
buy1 = ttk.Button(content, text="Buy", command=lambda: buy(stk1,stk1q))
buy2 = ttk.Button(content, text="Buy", command=lambda: buy(stk2,stk2q))
buy3 = ttk.Button(content, text="Buy", command=lambda: buy(stk3,stk3q))
buy4 = ttk.Button(content, text="Buy", command=lambda: buy(stk4,stk4q))

sell1 = ttk.Button(content, text="Sell", command=lambda: sell(stk1,stk1q))
sell2 = ttk.Button(content, text="Sell", command=lambda: sell(stk2,stk2q))
sell3 = ttk.Button(content, text="Sell", command=lambda: sell(stk3,stk3q))
sell4 = ttk.Button(content, text="Sell", command=lambda: sell(stk4,stk4q))

next= ttk.Button(content, text="Next Turn", command=lambda: next_turn())

#-------Establishing the Grid for the buttons and labels-------
content.grid(column=0, row=0, sticky=(N, S, E, W))

#-------Placing the News and a Frame around it-------
frame.grid(column=0, row=0, columnspan=5, rowspan=1, sticky=(N, S, E, W))
news.grid(column=0, row=0, columnspan=5, rowspan =1, sticky=(N, S, E, W), pady=20, padx=20)

#-------Creating the table for the stocks-------
namelbl.grid(column=0,row=1,sticky=(N, S, E, W))
stk1n.grid(column=0,row=2,sticky=(N, S, E, W))
stk2n.grid(column=0,row=3,sticky=(N, S, E, W))
stk3n.grid(column=0,row=4,sticky=(N, S, E, W))
stk4n.grid(column=0,row=5,sticky=(N, S, E, W))

pricelbl.grid(column=1,row=1,sticky=(N, S, E, W))
stk1p.grid(column=1,row=2,sticky=(N, S, E, W))
stk2p.grid(column=1,row=3,sticky=(N, S, E, W))
stk3p.grid(column=1,row=4,sticky=(N, S, E, W))
stk4p.grid(column=1,row=5,sticky=(N, S, E, W))

quantlbl.grid(column=2,row=1,sticky=(N, S, E, W))
stk1q.grid(column=2,row=2,sticky=(N, S, E, W))
stk2q.grid(column=2,row=3,sticky=(N, S, E, W))
stk3q.grid(column=2,row=4,sticky=(N, S, E, W))
stk4q.grid(column=2,row=5,sticky=(N, S, E, W))

choicelbl.grid(column=3,row=1,columnspan=2,sticky=(N, S, E, W))

buy1.grid(column=3,row=2,sticky=(N, S, E, W))
buy2.grid(column=3,row=3,sticky=(N, S, E, W))
buy3.grid(column=3,row=4,sticky=(N, S, E, W))
buy4.grid(column=3,row=5,sticky=(N, S, E, W))

sell1.grid(column=4,row=2,sticky=(N, S, E, W))
sell2.grid(column=4,row=3,sticky=(N, S, E, W))
sell3.grid(column=4,row=4,sticky=(N, S, E, W))
sell4.grid(column=4,row=5,sticky=(N, S, E, W))

#-------Placing the labels/button at the bottom-------
errorlbl.grid(column=0,row=6,columnspan=3,sticky=(N, S, E, W))
capitallbl.grid(column=3,row=6,columnspan=1,sticky=(N, S, E, W))
next.grid(column=4,row=6,columnspan=1,sticky=(N, S, E, W))

#-------Configuring the columns and rows to scale-------
root.columnconfigure(0, weight=1)
root.rowconfigure(0, weight=1)

content.rowconfigure(0, weight=6)
for i in range(5):
   content.columnconfigure(i, weight=1)
for i in range(1,6):
   content.rowconfigure(i, weight=1)

#-------Running the window-------
root.mainloop()